lib/galaxy/tools/imp_exp/imp_history_from_archive.xml

```diff
@@ -1,7 +1,11 @@
 <tool id="__IMPORT_HISTORY__" name="Import History" version="0.1" tool_type="import_history">
   <type class="ImportHistoryTool" module="galaxy.tools"/>
   <action module="galaxy.tools.actions.history_imp_exp" class="ImportHistoryToolAction"/>
-  <command>python '$__tool_directory__/unpack_tar_gz_archive.py' '${ str( $__ARCHIVE_SOURCE__ ).encode( 'base64' ) }' '${ str( $__DEST_DIR__ ).encode( 'base64' ) }' --$__ARCHIVE_TYPE__ --encoded</command>
+  <command>#from base64 import b64encode#
+python '$__tool_directory__/unpack_tar_gz_archive.py'
+'${ b64encode(bytearray(str($__ARCHIVE_SOURCE__), encoding="utf-8")).decode('utf-8')}'
```

Comment:
The need to wrestle with b64encode, bytearray and string decoding unfortunately made we realise that of course @jmchilton was right at the GCC , we probably need to keep supporting python 2 for a very long time if we want to remain reproducible. I think it's a small minority of tools though that will fail if not updated:(.